legate/core/operation.py

```diff
@@ -220,6 +270,23 @@ def get_name(self) -> str:
     def declare_partition(
         self, store: Store, disjoint: bool = True, complete: bool = True
     ) -> PartSym:
+        """
+        Creates a partition symbol for the store
+
+        Parameters
+        ----------
+        store : Store
+            Store to associate the partition symbol with
+        disjoint : bool, optional
+            ``True`` (by default) means the partition must be disjoint
+        complete : bool, optional
```

Comment:
Question rather than comment on the PR - but when would you ever want an incomplete partition? I get that disjoint = false might save you communication if you replicate some points, but seems weird to me that you would be okay with a partition that doesn't cover every point.  

Unless I'm misunderstanding complete.... and if I am, maybe we could clarify what complete means : )

Reply:
Partitions in legate and legion are not mathematical partitions, but a collection of subsets of a set. so the union of those subsets may not cover the original set. the runtime can and does perform a few optimizations when it knows that the partition used in updating a store is complete, so if a partition that you said is complete was actually incomplete, you would see correctness issues. lying in the other direction is actually safe, but will cause unnecessary runtime overhead. 

Incomplete partitions are useful when you know you only need to make accesses to a subset of data points, such as stencil accesses. A stencil code often creates two partitions of the same grid data, one that is complete and disjoint, used for updating the grid, and one that only selects elements used in stencil accesses. Doing this helps minimize the amount of data movement.

Reply:
Stencils, duh. Thank you.  LGTM.